lib/jekyll-relative-links/generator.rb

```diff
@@ -6,7 +6,7 @@ class Generator < Jekyll::Generator
     include Jekyll::Filters::URLFilters
 
     LINK_TEXT_REGEX = %r!([^\]]+)!
-    FRAGMENT_REGEX = %r!(#.+)?!
+    FRAGMENT_REGEX = %r!(#(?:[^)]+|.+$))?!
```

Comment:
The simple diff which I first proposed in #16 was:

```diff
-    FRAGMENT_REGEX = %r!(#.+)?!
+    FRAGMENT_REGEX = %r!(#[^)]+)?!
```

That is, making it non-greedy by simply requiring the whole of the fragment not to have a closing bracket.

That, however, breaks the test case I introduced in b52cca9: reference links with bracketed fragments.

Is up to you if you care about this last case. Seemed fair to fix it to me.

So the regex is:

  - EITHER a fragment with no `)`
  - OR if it has a `)`, consume the whole line — which files for inline links, since the final `)` would not match